cli/src/migrate.js

```diff
@@ -0,0 +1,354 @@
+'use strict';
+const chalk = require('chalk');
+const r = require('rethinkdb');
+const Promise = require('bluebird');
+const argparse = require('argparse');
+const child_process = require('child_process');
+const runSaveCommand = require('./schema').runSaveCommand;
+const config = require('./utils/config');
+const interrupt = require('./utils/interrupt');
+const change_to_project_dir = require('./utils/change_to_project_dir');
+const start_rdb_server = require('./utils/start_rdb_server');
+
+const VERSION_2_0 = [ 2, 0, 0 ];
+
+function run(cmdArgs) {
+  const options = processConfig(cmdArgs);
+  interrupt.on_interrupt(() => teardown());
+  return Promise.resolve().bind({ options })
+    .then(setup)
+    .then(validateMigration)
+    .then(makeBackup)
+    .then(renameUserTables)
+    .then(moveInternalTables)
+    .then(renameIndices)
+    .then(rewriteHzCollectionDocs)
+    .then(exportNewSchema)
+    .then(teardown)
+    .catch((err) => {
+      red(` └── ${err.message}`);
+      return Promise.bind(options).then(teardown());
```

Comment:
It wasn't doing cleanup for whatever reason, but I guess I can use `finally` for that